sqlmesh/core/snapshot/evaluator.py

```diff
@@ -1522,3 +1551,30 @@ def _intervals(snapshot: Snapshot, deployability_index: DeployabilityIndex) -> I
         if deployability_index.is_deployable(snapshot)
         else snapshot.dev_intervals
     )
+
+
+def _check_destructive_schema_change(
+    snapshot: Snapshot,
+    alter_expressions: t.List[exp.AlterTable],
+    allow_destructive_snapshots: t.Set[str],
+) -> None:
+    if (
+        snapshot.is_model
+        and snapshot.is_forward_only
+        and snapshot.name not in allow_destructive_snapshots
+        and not snapshot.model.on_destructive_change.is_ignore
```

Comment:
i feel like i've seen these conditions chained together multiple times, maybe this means we should have a method on the snapshot that computes this similar logic?